t3f/ops.py

```diff
@@ -1132,3 +1136,50 @@ def gather_nd(tt, indices):
     tt_elements = tf.matmul(tt_elements, core_slices)
   tt_elements = tf.reshape(tt_elements, tf.shape(indices)[:-1])
   return tt_elements
+
+def renormalize_cores(tt):
+    """Renormalizes tt cores to make them of the same Frobenius norm.
+
+    Does not change tt, but makes computations with cores more stable.
+
+    Args:
+      tt: `TensorTrain` or `TensorTrainBatch` object
+
+    Returns:
+      `TensorTrain` or `TensorTrainBatch` which represents the same
+      tensor as tt, but with all cores having equal norm. In the batch
+      case applies to each TT in `TensorTrainBatch`.
+
+    """
+    if isinstance(tt, TensorTrain):
+      new_cores = []
+      running_log_norm = 0
+      core_norms = []
+      for core in tt.tt_cores:
+        cur_core_norm = tf.sqrt(tf.reduce_sum(core ** 2) + 1e-8)
```

Comment:
Why do you need this eps, because of the gradient? Please, add it to the arguments (with 1e-8 being default value), make sure that this default value works with tf.float32, and think if we can avoid having eps whatsoever :)